### DISCUSSION TITLE
Import objects into Blender based on data provided by MTA:SA`s .map file

### DISCUSSION THREAD

**Discussion by riizom** [Parik27/DragonFF#338] [Ideas]
For many mappers and recorders it would be a great tool to work with.
the implemented logic coule be smth like this:

The user must have exported every .dff and every .png from every .txd and put it into one folder as preparation. There are many tools to do it in a couple minutes.
After installing and enabling DragonFF you would have to choose a folder in the preference tab which serves as directory for every exported file.
Under File -> Import you would find an option to import .map files. Those are getting read/analyzed for specific keywords.

Some created object in the MTA:SA editor look in the .map file like this:

![Screenshot 2024-09-10 204339](https://github.com/user-attachments/assets/b12f21b1-a822-4bfc-b4e2-842e05d15a6a)

Therefore filtering every line for keywords like "model" or posX, ... wouldnt be a problem. Since creating markers ingame would lead a line of that XML code to start with <marker instead of <object, the feature could filter for <object lines first and take every pos, rot, model and scale. For each line that contains <object at the beginning, the feature then would import an object from the folder based on the model id.
Working with a mapping dictionary which assigns ids to names will work. Here the model "8947" would have the name "vgElkup". Therefore the addon would search for a vgElkup.dff file and import it at the coordinates given by posX, posY and posZ with the same rotation given by rotX, rotY and rotZ. Since .pngs and .dff files are in the same folder, then imported object would automatically recognize the needed .png and assign it to itself. Ive tried to create it myself, but blender always imports every object provided by any .map file to the world origin of blender. Your .ipl/.ide importer does a great job here when importing map parts of GTA:SA since it imports every object to a correct position. is it possible to do so with a .map file?

**Comment by ghost** [comment]
There is no point in adding this to the plugin, as there is already an online IPL > .map converter made by the community on the internet.
